packages/fax/src/rest/v3/faxes/faxes-api.ts

```diff
@@ -0,0 +1,285 @@
+   * This endpoint supports the following content types for the fax payload:
+   *   - Multipart/form-data
+   *   - Application/json
+   * We will however always return a fax object in the response in application json.
+   * If you supply a callbackUrl the callback will be sent as multipart/form-data with the content of the fax as an attachment to the body, *unless* you specify callbackContentType as application/json.
+   * #### file(s)
+   *    Files may be included in the POST request as multipart/form-data parts.
+   * #### contentUrl
+   *    Any URL on the Internet (including ones with basic authentication), and we'll pull it down and make it a fax. This might be useful to you if you're using a web framework for templates and creating fax files.
+   *    Please note: If you are passing fax a secure URL (starting with 'https://'), make sure that your SSL certificate (including your intermediate cert, if you have one) is installed properly, valid, and up-to-date.
+   * @param { SendFaxRequestData } data - The data to provide to the API call.
+   */
+  public async send(data: SendFaxRequestData): Promise<Fax> {
+    this.client = this.getSinchClient();
+    const requestBody = data.sendFaxRequestBody;
+    requestBody['headerText'] = requestBody['headerText'] !== undefined
+      ? requestBody['headerText'] : '';
+    requestBody['headerPageNumbers'] = requestBody['headerPageNumbers'] !== undefined
+      ? requestBody['headerPageNumbers'] : true;
+    requestBody['headerTimeZone'] = requestBody['headerTimeZone'] !== undefined
+      ? requestBody['headerTimeZone'] : 'America/New_York';
+    requestBody['retryDelaySeconds'] = requestBody['retryDelaySeconds'] !== undefined
+      ? requestBody['retryDelaySeconds'] : 60;
+    requestBody['callbackContentType'] = requestBody['callbackContentType'] !== undefined
+      ? requestBody['callbackContentType'] : 'multipart/form-data';
+    requestBody['imageConversionMethod'] = requestBody['imageConversionMethod'] !== undefined
+      ? requestBody['imageConversionMethod'] : 'HALFTONE';
+    const getParams = this.client.extractQueryParams<SendFaxRequestData>(data, [] as never[]);
+    const headers: { [key: string]: string | undefined } = {
+      'Accept': 'application/json',
+    };
+
+    let body: RequestBody;
+    // Except if the request body contains a non-empty property 'files' (where the message will be sent as application/json)
+    // the request will be sent as multipart/formdata
+    const isUsingJson = (data['sendFaxRequestBody'] as FaxRequestJson).files !== undefined;
+    if (isUsingJson) {
+      headers['Content-Type'] = 'application/json';
+      body = JSON.stringify(data['sendFaxRequestBody']);
+    } else {
+      const formParams:any = {};
+      const requestData = data.sendFaxRequestBody as FaxRequestFormData;
+      if( requestData.to ) { formParams['to'] = requestData.to; }
+      if( requestData.file ) { formParams['file'] = requestData.file; }
```

Comment:
this faxrequest body contains a ton of readonly fields you can not set, price, barcodes etc please refer to https://developers.sinch.com/docs/fax/api-reference/fax/tag/Faxes/#tag/Faxes/operation/sendFax for valid in parameters.

**Question** : where are the files proccessed for multpard and applicaiton json, (base64 for json.
**Question 2**: can you show me an example on how its sent? and could we have a few money maker methods for docs and the normal use

example 
sendFax(to,from,file)
sendFax(to,from,contenturl)

In the sdk im not sure we want to expose that a customer can choose json or form data, i think it would be better that we always post multipart formdata its more effective and hide that fact for teh sdk user. no need to complicate it

Reply:
Then it would be good to not define those fields as part of the request body in the documentation: at the moment it mentions:
 - to (required)
 - file (optional)
 - from (optional)
 - contentUrl (optional)
 - headerText (optional) - default: ""
 - headerPageNumbers (optional) - default: true
 - headerTimeZone (optional) - default: "America/New_York"
 - retryDelaySeconds (optional) - default: 60
 - labels (optional)
 - callbackUrl (optional)
 - callbackContentType (optional) - default "multipart/form-data"
 - imageConversionMethod (optional) - default "HALFTONE"
 - serviceId (optional)
 - maxRetries (optional)
 The SDK is not meant to downgrade the possibilities offered by the API so all these fields are exposed. If they are not meant to be exposed to the customer, then please update the specification accordingly, I'll then happily update the request interface.

Reply:
> Question : where are the files proccessed for multpard and applicaiton json, (base64 for json.

I'm not sure to understand your question, can you please clarify?

Reply:
> Then it would be good to not define those fields as part of the request body in the documentation: at the moment it mentions:
> 
> * to (required)
> * file (optional)
> * from (optional)
> * contentUrl (optional)
> * headerText (optional) - default: ""
> * headerPageNumbers (optional) - default: true
> * headerTimeZone (optional) - default: "America/New_York"
> * retryDelaySeconds (optional) - default: 60
> * labels (optional)
> * callbackUrl (optional)
> * callbackContentType (optional) - default "multipart/form-data"
> * imageConversionMethod (optional) - default "HALFTONE"
> * serviceId (optional)
> * maxRetries (optional)
>   The SDK is not meant to downgrade the possibilities offered by the API so all these fields are exposed. If they are not meant to be exposed to the customer, then please update the specification accordingly, I'll then happily update the request interface.

Compare that to the current request in node and you will see for example price is a request parameter and it should not be and it’s not in the docs. It looks the request is infact the complete faxobject

Reply:
> > In the sdk im not sure we want to expose that a customer can choose json or form data,
> 
> Again, the SDK doesn't aim to restrict the API possibilities. According to what I see in the documentation, the only contentType default is the one for the callback. If the Fax API PM want to promote formdata as a default request, this needs to be documented and communicated to us so that we act accordingly in the SDK.

I can confirm as the api manager that I would like to promote formencided over json when sending faxes just because it’s more efficient

Reply:
> > could we have a few money maker methods for docs and the normal use
> 
> This has to be discussed with the SDK team to decide how to present such method, versus the complete ones

Ok I see similar need in sms tbh especially in documentation we need to beat competition in lines of code and clarity

Reply:
> Compare that to the current request in node and you will see for example price is a request parameter and it should not be and it’s not in the docs. It looks the request is infact the complete faxobject

I see it now, I've removed them. Thanks

Reply:
> I can confirm as the api manager that I would like to promote formencided over json when sending faxes just because it’s more efficient

> Ok I see similar need in sms tbh especially in documentation we need to beat competition in lines of code and clarity

Interesting topics! 👍 For both these comments, I'll let you schedule a meeting with our manager to discuss further and plan accordingly

Reply:
> > Compare that to the current request in node and you will see for example price is a request parameter and it should not be and it’s not in the docs. It looks the request is infact the complete faxobject
> 
> I see it now, I've removed them. Thanks

you removed all, like pagecount, status etc?

Reply:
I've updated the fax request model to match strictly with the online documentation: the diff can be seen in the commit https://github.com/sinch/sinch-sdk-node/pull/32/commits/6691399e2229a9a1a564e989fbfa68ad58a6568a